pinot-common/src/test/java/org/apache/pinot/common/function/JsonPathMapCacheTest.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.function;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Predicate;
+import com.jayway.jsonpath.internal.Utils;
+import com.jayway.jsonpath.spi.cache.Cache;
+import com.jayway.jsonpath.spi.cache.CacheProvider;
+import java.util.Collections;
+import java.util.LinkedList;
+import org.apache.pinot.common.function.scalar.JsonFunctions;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class JsonPathMapCacheTest {
+
+  private String _jsonString;
+
+  @BeforeMethod
+  public void beforeMethod() {
+    _jsonString = "{\"name\":\"Jack\",\"contact\":{\"phone\":\"01234567\",\"email\":\"test@example.com\"}}";
+  }
+
+  @Test
+  public void testSimpleJsonPathMapCacheUsed() throws JsonProcessingException {
+    // request json path function, verify SimpleJsonPathMapCache used
+    assertEquals(JsonFunctions.jsonPathString(_jsonString, "$.contact.email"), "test@example.com");
+    Cache cache = CacheProvider.getCache();
+    assertEquals(cache.getClass(), JsonPathMapCache.class);
+  }
+
+  @Test
+  public void testSimpleJsonPathMapCacheWorks() throws JsonProcessingException {
+    String path = "$.contact.email";
+    assertEquals(JsonFunctions.jsonPathString(_jsonString, path), "test@example.com");
+    JsonPathMapCache cache = (JsonPathMapCache) CacheProvider.getCache();
+
+    // verify json path has been cached
+    LinkedList<Predicate> filterStack = new LinkedList<>(Collections.emptyList());
+    String cacheKey = Utils.concat(path, filterStack.toString());
+    JsonPath jsonPath = cache.get(cacheKey);
+    assertNotNull(jsonPath);
+  }
```

Comment:
This depends on undocumented library behaviour, it assumes the format of the cache keys. I have actually submitted a PR to the library to change this behaviour, which would break this test: https://github.com/json-path/JsonPath/pull/750

I suggest iterating over the map instead (requires exposing the keys on `JsonPathMapCache`).

Reply:
Or just use a Mockito spy and verify `put` was called, I guess.

Reply:
Good suggestions. 
I always prefer to use Mockitio for unit testing, but all method calls related to JSONFunctions are static, so I try to check the map size now.